src/Data/AbstractParams.php

```diff
@@ -13,24 +13,26 @@
  */
 abstract class AbstractParams implements JsonSerializable
 {
+    protected array $params = [];
+
     final public function __construct(
-        protected array $params = []
+        array $params = []
     ) {
-        //
+        $this->params = $params;
     }
 
     /**
      * Obtain a new instance.
      */
-    public static function new(array $params = []): static
+    public static function new(array $params = []): self
```

Comment:
With these i think we need an @return docblock to indicate it actually returns static, not self